Sources/NIOKit/FutureExtensions.swift

```diff
@@ -0,0 +1,13 @@
+// MARK: - Guard
+
+public extension EventLoopFuture {
+	/// Guards that the future's value satisfies the callback's condition or fails with the given error.
+	func `guard`(_ callback: @escaping ((Value) -> Bool), else error: Error) -> EventLoopFuture<Value> {
```

Comment:
The error that is passed in should be marked `@autoclosure`. This prevents it from bring initialized unless it is actually used.